src/scripts/underground/DailyDeal.ts

```diff
@@ -39,7 +39,35 @@ class DailyDeal {
     }
 
     private isValid(dealList: Array<DailyDeal>): boolean {
-        return ( (this.item1.name !== this.item2.name) && !DailyDeal.reverseDealExists(this.item1.name, this.item2.name, dealList) && !this.item1.isStone() );
+        const item1Name = this.item1.name;
+        const item2Name = this.item2.name;
+
+        if (item1Name !== item2Name) {
+            return false;
+        }
+
+        if (!this.item1.isStone()) {
```

Comment:
I _think_ this might be the wrong way around?
Although I am a bit tired, so I may be wrong 😓
```suggestion
        if (this.item1.isStone()) {
```